pkgs/by-name/do/dorion/package.nix

```
@@ -1,65 +1,190 @@
 { lib
 , stdenv
+, fetchFromGitHub
+, fetchYarnDeps
 , fetchurl
+, rustPlatform
 , autoPatchelfHook
-, dpkg
+, wrapGAppsHook
+, esbuild
 , glib-networking
 , gst_all_1
 , libappindicator
 , libayatana-appindicator
+, nodejs
+, openssl
+, pkg-config
+, pnpm-lock-export
+, prefetch-yarn-deps
 , webkitgtk
-, wrapGAppsHook
+, yarn
 }:
 
-stdenv.mkDerivation (finalAttrs: {
-  name = "dorion";
-  version = "1.2.1";
+let
+
+  pname = "dorion";
+  version = "2.2.1";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-FghJM34GMt8+4b6jsQQSsfmHIyua/pjRHKNErGyK/kw=";
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-MlpkbK1CoU1gOst/qH+hhJPhKRJacpdodhbl34XNcYM=";
+
+    nativeBuildInputs = [ pnpm-lock-export ];
+    postFetch = ''
+      (cd $out && pnpm-lock-export --schema yarn.lock@v1)
+    '';
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/ea1db7b8780b56b9dabf65a0ea7bd87a0265907e/shelter.js";
+    hash = "sha256-pH+zc7ck6rP3F3UQZHTtORYYvEH6HAbI2bJYcKaCFUU=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ];
+      license = lib.licenses.cc0;
+    };
+  };
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
-  ];
+  frontend = stdenv.mkDerivation {
+    pname = "dorion-ui";
+    inherit version src;
+
+    offlineCache = fetchYarnDeps {
+      yarnLock = "${src}/yarn.lock";
+      hash = "sha256-tEIaEQqnnrbJMMPICVAdU0EQ6AxhLJ3xhGczUQoG1W0=";
+    };
```

On my end the current build failure on ofborg is resolved by this fix.

```suggestion
      NODE_EXTRA_CA_CERTS = "${cacert}/etc/ssl/certs/ca-bundle.crt";
    };
```

---

Done

---

@lilyinstarlight, i'm pinging you to put this fix on your radar.